apps/emqx_management/test/emqx_mgmt_api_clients_SUITE.erl

```diff
@@ -171,6 +215,290 @@ t_clients(_) ->
 
+t_persistent_sessions1(Config) ->
+    [N1, _N2] = ?config(nodes, Config),
+    APIPort = 18084,
+    Port1 = get_mqtt_port(N1, tcp),
+
+    ?assertMatch({ok, {{_, 200, _}, _, #{<<"data">> := []}}}, list_request(APIPort)),
+
+    ?check_trace(
+        begin
+            %% Scenario 1
+            %% 1) Client connects and is listed as connected.
+            %% 2) Client disconnects and is listed as disconnected.
+            %% 3) Client reconnects and is listed as connected.
+            %% 4) Client disconnects.
+            %% 5) Session is GC'ed, client is removed from list.
+            ?tp(notice, "scenario 1", #{}),
+            O = #{api_port => APIPort},
+            ClientId = <<"c1">>,
+            C1 = connect_client(#{port => Port1, clientid => ClientId}),
+            assert_single_client(O#{node => N1, clientid => ClientId, status => connected}),
+            ok = emqtt:disconnect(C1),
+            assert_single_client(O#{node => N1, clientid => ClientId, status => disconnected}),
+            C2 = connect_client(#{port => Port1, clientid => ClientId}),
+            assert_single_client(O#{node => N1, clientid => ClientId, status => connected}),
+            ok = emqtt:stop(C2),
+            %% simulate GC
+            ?tp(notice, "gc", #{}),
+            ok = erpc:call(N1, emqx_persistent_session_ds, destroy_session, [ClientId]),
+            ?retry(
+                100,
+                20,
+                ?assertMatch(
+                    {ok, {{_, 200, _}, _, #{<<"data">> := []}}},
+                    list_request(APIPort)
+                )
+            ),
+            ok
+        end,
+        []
+    ),
+    ok.
+
+t_persistent_sessions2(Config) ->
+    [N1, _N2] = ?config(nodes, Config),
+    APIPort = 18084,
+    Port1 = get_mqtt_port(N1, tcp),
+
+    ?assertMatch({ok, {{_, 200, _}, _, #{<<"data">> := []}}}, list_request(APIPort)),
+
+    ?check_trace(
+        begin
+            %% Scenario 2
+            %% 1) Client connects and is listed as connected.
+            %% 2) Client connects to the same node and takes over, listed only once.
+            ?tp(notice, "scenario 2", #{}),
+            O = #{api_port => APIPort},
+            ClientId = <<"c2">>,
+            C1 = connect_client(#{port => Port1, clientid => ClientId}),
+            assert_single_client(O#{node => N1, clientid => ClientId, status => connected}),
+            unlink(C1),
+            C2 = connect_client(#{port => Port1, clientid => ClientId}),
+            assert_single_client(O#{node => N1, clientid => ClientId, status => connected}),
+            ok = emqtt:stop(C2),
+            ok = erpc:call(N1, emqx_persistent_session_ds, destroy_session, [ClientId]),
+            ?retry(
+                100,
+                20,
+                ?assertMatch(
+                    {ok, {{_, 200, _}, _, #{<<"data">> := []}}},
+                    list_request(APIPort)
+                )
+            ),
+
+            ok
```

Comment:
Nit: not needed?

Reply:
Those `ok`s are just for ease of adding new lines without adding commas to the diff.  e.g.: `?assert(...).` would become `?assert(...),` just to add a newline.

I think those `ok`s are innocuous, hence I prefer to leave them there.

Reply:
Not sure if we have to optimize code for "minimal diffs". I would prefer to optimize for "readability at any given moment".